File: Sum Root to Leaf Numbers.cpp
struct TreeNode {
     int val;
     TreeNode *left;
     TreeNode *right;
     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
};
 

class Solution {
public:
    int sumNumbers(TreeNode* root) {
        this->sumNumbers(root, 0);
    }

    int sumNumbers(TreeNode *root, int prefix){
        if (!root) return 0;
        if (!root->left) && (!root->right){
            return prefix * 10 + root->val;
        }
        int ret = 0, my_prefix = prefix * 10 + root->val;
        if (root->left) ret += this->sumNumbers(root->left, my_prefix);
        if (root->right) ret += this->sumNumbers(root->right, my_prefix);
        return ret
    }
};

File: Sum Root to Leaf Numbers.py
from binary_tree import Tree, TreeNode

class Solution:
    def sumNumbers(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        res = 0
        def sumTree(root, s):
            nonlocal res
            if not root:return
            if (not root.left) and (not root.right):
                res += int(s + str(root.val))
            else:
                if root.left:sumTree(root.left, s + str(root.val))
                if root.right:sumTree(root.right, s + str(root.val))
        sumTree(root, '')
        return res

print(Solution().sumNumbers(Tree([]).root))
